### PR TITLE
Support run/plan jobfile via URL location

### PR DESCRIPTION
**Pull request by nak3**

### CODE REVIEW

**Comment by nak3** [comment]
Suppot jobfile via URL like below:

~~~
$  nomad run https://raw.githubusercontent.com/nak3/nomad-playground/master/jobs/nginx.nomad
==> Monitoring evaluation "77543788"
    Evaluation triggered by job "nginx"
    Allocation "4d5af5c7" created: node "d9525e74", group "cache"
    Evaluation status changed: "pending" -> "complete"
==> Evaluation "77543788" finished with status "complete"
~~~

**Comment by dadgar** [comment]
This is cool! I have three requests:

1) Lets use [go-getter](https://github.com/hashicorp/go-getter). This will let us download jobs in a bunch of ways.

2) Lets support `nomad validate`

3) Add documentation to the commands and website explaining that we support downloading anything in go-getter syntax

**Comment by nak3** [comment]
Thank you. I added 2) support `nomad validate`.

But please let me discuss about using go-getter.
Current implentation doesn't download the job, but use HTTP response body like [this](https://github.com/nak3/nomad/blob/c70075c4f8b5ab66508b836cfb5c1bcc5204e63c/command/plan.go#L152).

While, if we use go-getter with supported protocol (e.g git/hg), it will not download only one jobfile, but git/hg clone entire repository. So, I feel it is not efficient.

(For example...) If I try to use [this git project file](https://github.com/nak3/nomad-playground/blob/master/jobs/nginx.nomad) with go-getter, it will clone entire [git project](https://github.com/nak3/nomad-playground) and specify the one file.

**Comment by dadgar** [comment]
Hey @nak3,

In the specific case that you give it a git repo, you are correct that it will download the whole thing. However, if you give it the raw url to the file in github like (https://raw.githubusercontent.com/nak3/nomad-playground/master/jobs/nginx.nomad) it will download it using HTTP. By using go-getter you can download the file from more locations which is nice.

We should also refactor the code into a helper so that we don't duplicate everywhere

**Comment by nak3** [comment]
@dadgar OK, I understand. I update them to use `go-getter`. Thank you.

File: command/helpers.go [hunk 1]
```diff
@@ -216,3 +222,63 @@ READ:
 	// Just stream from the underlying reader now
 	return l.ReadCloser.Read(p)
 }
+
+type Helper struct {
```
Comment (dadgar) [inline command/helpers.go]: Lets change the name to `JobGetter` and add a comment on the struct

File: command/helpers_test.go [hunk 2]
```diff
@@ -185,3 +186,42 @@ func TestHelpers_LineLimitReader_TimeLimit(t *testing.T) {
...
+	if err != nil {
+		t.Fatalf("err: %s", err)
+	}
+
+	err = sj.Validate()
+	if err != nil {
+		t.Fatalf("err: %s", err)
+	}
+
+}
```
Comment (dadgar) [inline command/helpers_test.go]: Can add a test that host the file using an HTTP server

File: website/source/docs/commands/validate.html.md.erb [hunk 3]
```diff
@@ -19,7 +19,11 @@ nomad validate <file>
 
 The validate command requires a single argument, specifying the path to a file
 containing a [HCL job specification](/docs/jobspec/index.html). This file
-will be read and the job checked for any problems.
+will be read and the job checked for any problems. If the
+supplied path is "-", the jobfile is read from STDIN. Otherwise it is read
+from the file at the supplied path or downloaded and read from URL specified.
+Nomad downloads jobfile using [`go-getter`](https://github.com/hashicorp/go-getter)
+and support `go-getter` syntax.
```
Comment (dadgar) [inline website/source/docs/commands/validate.html.md.erb]: Can we add this to plans documentation as well

**Comment by dadgar** [comment]
Small comments but LGTM

**Comment by nak3** [comment]
Thank you. I updated them.

**Comment by dadgar** [comment]
@nak3 Looks like the tests are failing

**Comment by nak3** [comment]
@dadgar I don't think test was failed due to this PR. I fetched and re-push the PR, and now it passed tests.

**Comment by dadgar** [comment]
Ah... The travis tests misreport their success right now. Working on that. Here are the raw logs:
https://s3.amazonaws.com/archive.travis-ci.org/jobs/152858750/log.txt

TestPlanCommand_From_URL
TestRunCommand_Fails
....

**Comment by nak3** [comment]
Oh, sorry. I missed it. Thank you. I updated.

**Comment by dadgar** [comment]
Awesome thank you so much!

**Pull request merged**